File: Contracts/Model/Product/Response/OptionDetailResponse.cs
using Contracts.Model.Product.UserInteract.Response;

namespace Contracts.Model.Product.Response;
public class OptionDetailResponse
{
	public Guid Id { get; set; }
	public string Title { get; set; }
	public string? Description { get; set; }
	public ICollection<MultipleImagesResponse>? Images { get; set; }
	public MultipleImagesResponse Image { get; set; }
	public Guid ImageId { get; set; }
	public double Price { get; set; }
	public string? Color { get; set; }
	public string GenderFor { get; set; }
	public string Stock => StockCalculate().ToString();
	public int QuantityAvailable { get; set; }
	public DiscountPreviewResponse? Discount { get; set; }
	public ICollection<ReviewResponse>? Reviews { get; set; }
	public string Condition { get; set; }
	private StockStatusType StockCalculate()
	{
		if (QuantityAvailable == 0)
			return StockStatusType.OutOfStock;
		if (QuantityAvailable > 0 && QuantityAvailable < 10)
			return StockStatusType.LowStock;
		if (QuantityAvailable >= 10)
			return StockStatusType.InStock;
		return StockStatusType.InStock;
	}
}

File: ClientApp/src/app/core/interfaces/Product/Response/optionDetailResponse.ts
import { ReviewResponse } from "../UserInteract/Response/reviewResponse";
import { DiscountPreviewResponse } from "./discountPreviewResponse";
import { MultipleImagesResponse } from "./multipleImagesResponse";

export interface OptionDetailResponse {
    id: string;
    title: string
    description: string | null;
    images: MultipleImagesResponse[] | null;
    image: MultipleImagesResponse;
    price: number;
    color: string | null;
    genderFor: string;
    stock: string;
    quantityAvailable: number;
    offerPrice: number | null;
    discount: DiscountPreviewResponse | null;
    averageRating: number;
    totalReviews: number;
    reviews: ReviewResponse[] | null;
    condition: string;
}